# Logistic Regression?

In [20]:
import pandas as pd
from sklearn.model_selection import train_test_split
from sklearn.preprocessing import MinMaxScaler

df = pd.read_csv('turtles.csv')
df = df.dropna(how='any')
y = df.SPECIES
X = df[['WEIGHTGR', 'LENGTHCM', 'WIDTHCM', 'DEPTHCM']]

X_train, X_test, y_train, y_test = train_test_split(X, y, stratify=y)
X_train = MinMaxScaler().fit_transform(X_train)
X_test = MinMaxScaler().fit_transform(X_test)

In [21]:
y_test.value_counts() / len(y_test)

loggerhead      0.791064
kemps-ridley    0.208936
Name: SPECIES, dtype: float64

In [29]:
from sklearn.neural_network import MLPClassifier

mlp = MLPClassifier(hidden_layer_sizes=(1,),
                    activation='logistic' # just logistic regression at this point
                   ).fit(X_train, y_train) # essentially just logistic regression
mlp.score(X_test, y_test)

0.7910643889618922

In [27]:
from sklearn.linear_model import LogisticRegression

logit = LogisticRegression(fit_intercept=True, 
                           max_iter=200 # same as MLPClassifier
                          ).fit(X_train, y_train)

logit.score(X_test, y_test)

0.9369250985545335

Now that's interesting! An MLPClassifier with a single perceptron (so much for the ML in MLP) has exactly the same performance as the LogisticRegression model. I'm 79.106% sure that I've just screwed something up, though...

In [30]:
from sklearn.naive_bayes import GaussianNB

gnb = GaussianNB().fit(X_train, y_train)
gnb.score(X_test, y_test)

0.6044678055190539

Well... that was terrible.

In [31]:
from sklearn.tree import DecisionTreeClassifier

dt = DecisionTreeClassifier()
dt.fit(X_train, y_train)
dt.score(X_test, y_test)

0.8199737187910644